langchain/chains/conversation/base.py

```diff
@@ -39,6 +39,12 @@ def input_keys(self) -> List[str]:
         """Use this since so some prompt vars come from history."""
         return [self.input_key]
 
+    def __init__(self, **kwargs):
+        """Initialize the chain."""
+        super().__init__(**kwargs)
+        if self.memory is not None and self.memory.input_key is None:
+            self.memory.input_key = self.input_key
```

Comment:
```suggestion
def __init__(self, **kwargs):
    """Initialize the chain."""
    super().__init__(**kwargs)
    if self.memory is not None and self.memory.input_key is None:
        raise ValueError("Memory input key must be explicitly set when creating an instance of ConversationBufferMemory.")

```

Reply:
The memory key should be set explicitly by the user when creating an instance of `ConversationBufferMemory`, and not automatically set in the `__init__` method.